gap/PolygonalComplexes/animating.gd

```diff
@@ -0,0 +1,537 @@
+#! @EndGroup
+
+#! @BeginGroup SetEdgeColours
+#! @Description
+#! Set colours for all edges. (Colours are stored in the format 0xABCDEF where A,B,C,D,E,F are elements of the hexadecimal code.  For more information look at the start of the section <Ref Subsect="Section_LabelColouring"/>).
+#! @Returns the updated print record.
+#! @Arguments surface, newColoursList, printRecord
+DeclareOperation( "SetEdgeColours", [IsSimplicialSurface and IsNotEdgeRamified, IsDenseList, IsRecord] );
+#! @Arguments surface, printRecord
+DeclareOperation( "GetEdgeColours", [IsSimplicialSurface and IsNotEdgeRamified, IsRecord] );
+#! @EndGroup
+
+#! @BeginGroup SetEdgeColour
+#! @Description
+#! Set the <A>colour</A> of the edge in position <A>i</A> of <A>Edges(surface)</A>. (Colours are stored in the format 0xABCDEF where A,B,C,D,E,F are elements of the hexadecimal code.  For more information look at the start of the section <Ref Subsect="Section_LabelColouring"/>).
+#! @Returns the updated print record.
+#! @Arguments surface, i, colour, printRecord
+DeclareOperation( "SetEdgeColour", [IsSimplicialSurface and IsNotEdgeRamified, IsCyclotomic, IsString, IsRecord] );
+#! @Arguments surface, i, printRecord
+DeclareOperation( "GetEdgeColour", [IsSimplicialSurface and IsNotEdgeRamified, IsCyclotomic, IsRecord] );
+#! @EndGroup
+
+
+#! @BeginGroup SetFaceColours
+#! @Description
+#! Set colours for all faces. (Colours are stored in the format 0xABCDEF where A,B,C,D,E,F are elements of the hexadecimal code.  For more information look at the start of the section <Ref Subsect="Section_LabelColouring"/>).
+#! @Returns the updated print record.
+#! @Arguments surface, newColoursList, printRecord
+DeclareOperation( "SetFaceColours", [IsSimplicialSurface and IsNotEdgeRamified, IsDenseList, IsRecord] );
+#! @Arguments surface, printRecord
+DeclareOperation( "GetFaceColours", [IsSimplicialSurface and IsNotEdgeRamified, IsRecord] );
+#! @EndGroup
+
+#! @BeginGroup SetFaceColour
+#! @Description
+#! Set the <A>colour</A> of the face in position <A>i</A> of <A>Faces(surface)</A>. (Colours are stored in the format 0xABCDEF where A,B,C,D,E,F are elements of the hexadecimal code.  For more information look at the start of the section <Ref Subsect="Section_LabelColouring"/>).
+#! @Returns the updated print record.
+#! @Arguments surface, i, colour, printRecord
+DeclareOperation( "SetFaceColour", [IsSimplicialSurface and IsNotEdgeRamified, IsCyclotomic, IsString, IsRecord] );
+#! @Arguments surface, i, printRecord
+DeclareOperation( "GetFaceColour", [IsSimplicialSurface and IsNotEdgeRamified, IsCyclotomic, IsRecord] );
+#! @EndGroup
+
+#! @Section Inner Circles
+#! @SectionLabel LabelInnerCirclesAnimating
+#!
+
+#! For an inner circle we need a face with 3 corresponding edges. Then an inner circle is an circle which lie in the face such that the circle touches each edge of the face in exactly one point.
```

Comment:
Inner circles are defined for triangular faces. In the animation, they are circles within the face that touch each edge in exactly one point.